enhancements/network/communication-ingress-flows-matrix.md

```diff
@@ -0,0 +1,198 @@
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### Workflow Description
+
+An OpenShift administrator would like to get an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- The admin reviews OpenShift release documentation to get the included communication
+  matrix describing the expected flows of incoming traffic.
+
+- The admin uses the OpenShift command-line interface (CLI) to generate an up-to-date 
+  communication matrix using the following command `oc adm generate-commatrix`.
+
+- Optionally, the admin uses the OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+
+A new helper module `node-comm-lib` will be added to `openshift/library-go`. #Sabina: what should be the module name?
+The module will be used to create a communication ingress flows matrix 
+from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` and should be created 
+by the Machine API Operator during cluster deployment.
+
+Sabina to Yuval: What k8s link you wanted to add here?
+
+The module allows to generate the communication matrix in various formats:
+- YAML/JSON
+- CSV
+- nftables
+
+Each record describes a flow with the following information:
+```
+traffic-direction     Data flow direction (ingress/egress)
+protocol              Network protocol (TCP/UDP)
+port-number           Flow port number
+service-name          OpenShift Service name
+namespace             Service namespace
+node-role             Service node host role (control-plane/master/worker)
+optional              Optional or mandatory flow for OpenShift
+```
+
+For example,
+`ingress,TCP,18080,openshift-kni-infra-coredns-cnfdr8.telco5g.eng.rdu2.redhat.com-coredns,worker,true`
+
+### Risks and Mitigations
+
+For egress flows support (out of scope) an API change might be needed. 
+`EndpointSlices` stand better for ingress traffic than for egress, in ingress traffic the 
+services are less dynamic and mostly stay up during the entire cluster lifetime.
+Supporting egress traffic might require changes in the API that should be reviewed and 
+agreed with the upstream Kubernetes community.
```

Comment:
the egress risk - the risk is that when we'll get to do that part, which is currently out-of-scope, we might need to change the ingress part
especially considering the fact that we plan on bringing this upstream too